File: client/java/src/main/java/com/supremainc/sdk/example/user/UserSvc.java
package com.supremainc.sdk.example.user;

import java.util.List;

import com.supremainc.sdk.user.GetListRequest;
import com.supremainc.sdk.user.GetListResponse;
import com.supremainc.sdk.user.GetRequest;
import com.supremainc.sdk.user.GetResponse;
import com.supremainc.sdk.user.EnrollRequest;
import com.supremainc.sdk.user.EnrollResponse;
import com.supremainc.sdk.user.DeleteRequest;
import com.supremainc.sdk.user.DeleteResponse;
import com.supremainc.sdk.user.UserGrpc;
import com.supremainc.sdk.user.UserHdr;
import com.supremainc.sdk.user.UserInfo;
import com.supremainc.sdk.user.UserFinger;
import com.supremainc.sdk.user.SetFingerRequest;
import com.supremainc.sdk.user.SetFingerResponse;

public class UserSvc {
  private final UserGrpc.UserBlockingStub userStub;

  public UserSvc(UserGrpc.UserBlockingStub stub) {
    userStub = stub;
  }

  public List<UserHdr> getList(int deviceID) throws Exception {
    GetListRequest request = GetListRequest.newBuilder().setDeviceID(deviceID).build();
    GetListResponse response = userStub.getList(request);

    return response.getHdrsList();
  } 

  public List<UserInfo> getUser(int deviceID, List<String> userIDs) throws Exception {
    GetRequest request = GetRequest.newBuilder().setDeviceID(deviceID).addAllUserIDs(userIDs).build();
    GetResponse response = userStub.get(request);

    return response.getUsersList();
  }

  public void enroll(int deviceID, List<UserInfo> users) throws Exception {
    EnrollRequest request = EnrollRequest.newBuilder().setDeviceID(deviceID).addAllUsers(users).build();
    EnrollResponse response = userStub.enroll(request);
  }

  public void delete(int deviceID, List<String> userIDs) throws Exception {
    DeleteRequest request = DeleteRequest.newBuilder().setDeviceID(deviceID).addAllUserIDs(userIDs).build();
    DeleteResponse response = userStub.delete(request);
  }

  public void setFinger(int deviceID, List<UserFinger> userFingers) throws Exception {
    SetFingerRequest request = SetFingerRequest.newBuilder().setDeviceID(deviceID).addAllUserFingers(userFingers).build();
    SetFingerResponse response = userStub.setFinger(request);
  }
}

File: client/csharp/example/user/UserSvc.cs
using System;
using User;
using Grpc.Core;
using Google.Protobuf.Collections;

namespace example
{
  class UserSvc
  {
    private User.User.UserClient userClient;

    public UserSvc(Channel channel) {
      userClient = new User.User.UserClient(channel);
    }

    public RepeatedField<User.UserHdr> GetList(uint deviceID) {
      var request = new GetListRequest{ DeviceID = deviceID };

      try {
        var response = userClient.GetList(request);

        return response.Hdrs;
      } catch(RpcException e) {
        Console.WriteLine("Cannot get the user list {0}: {1}", deviceID, e);
        throw;
      }
    }

    public RepeatedField<User.UserInfo> GetUser(uint deviceID, string[] userIDs) {
      var request = new GetRequest{ DeviceID = deviceID };
      request.UserIDs.AddRange(userIDs);

      try {
        var response = userClient.Get(request);

        return response.Users;
      } catch(RpcException e) {
        Console.WriteLine("Cannot get the user information {0}: {1}", deviceID, e);
        throw;
      }
    } 

    public void Enroll(uint deviceID, User.UserInfo[] users) {
      var request = new EnrollRequest{ DeviceID = deviceID };
      request.Users.AddRange(users);

      try {
        userClient.Enroll(request);
      } catch(RpcException e) {
        Console.WriteLine("Cannot enroll users {0}: {1}", deviceID, e);
        throw;
      }
    }

    public void Delete(uint deviceID, string[] userIDs) {
      var request = new DeleteRequest{ DeviceID = deviceID };
      request.UserIDs.AddRange(userIDs);

      try {
        userClient.Delete(request);
      } catch(RpcException e) {
        Console.WriteLine("Cannot delete users {0}: {1}", deviceID, e);
        throw;
      }
    }

    public void SetFinger(uint deviceID, User.UserFinger[] userFingers) {
      var request = new SetFingerRequest{ DeviceID = deviceID };
      request.UserFingers.AddRange(userFingers);

      try {
        userClient.SetFinger(request);
      } catch(RpcException e) {
        Console.WriteLine("Cannot set fingers {0}: {1}", deviceID, e);        
        throw;
      }
    }   
  }
}